09/hw-09.py

```diff
@@ -0,0 +1,73 @@
+# 1. Створити програму, яка буде приймати число і друкувати відповідне число в послідовності Фібоначчі,
+#    використовуючи генератори.
+#
+# 2. Створити програму, яка буде приймати число і друкувати відповідне число в послідовності Фібоначчі,
+#    використовуючи ітератори (необов'язкове виконання).
+#
+# 3. Створити програму, яка буде приймати число і друкувати відповідне число в послідовності Фібоначчі,
+#    використовуючи рекурсію (необов'язкове виконання).
+#
+# 4. Написати програму, яка буде повертати факторіал введеного числа, використовуючи рекурсію (необов'язкове виконання).
+# ===============================================================
+
+# Task 1.
+def get_number_by_index_from_fibonacci_sequence(index):
+    a, b = 0, 1
+    for _ in range(index + 1):
+        yield a
+        a, b = b, a + b
+
+
+number = int(input('Enter your number: '))
+result = list(get_number_by_index_from_fibonacci_sequence(number))[number]
```

Comment:
you can use [-1] here to get the last element of the list

Reply:
Yeah, sure.